### PR TITLE
MM_33223 [v2] Added Prettier to the project

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
This PR adds Prettier to the v2 branch.  It will help in maintaining a standardized code style throughout the development process.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33223

### CODE REVIEW

**Comment by enahum** [comment]
I don't see a need for this... eslint dhould be enough

**Comment by avinashlng1080** [comment]
We do have some code formatting with Eslint but it is limited in some cases.  With Prettier, a developer does not need to spend time fixing them. 

It also helps writing better code.  For example, in the v2 app, the [line 6](https://github.com/mattermost/mattermost-mobile/blob/d6a3504c08aaf3165b36d44110fa45f321aea39f/types/api/client_error.d.ts#L6) of interface ClientErrorProps looks good : 
 `   intl?: {defaultMessage?: string; id: string;} | { defaultMessage?: string; id: string } | { id: string; defaultMessage?: string; values: any } | { id: string; defaultMessage?: string };`
 
However,  while inspecting the file changes made by Prettier, I noticed the duplicate ` { defaultMessage?: string; id: string } ` and I removed [it.](https://github.com/mattermost/mattermost-mobile/blob/37acfc1a8d550e5d8c214d9f067192bb6b334a3b/types/api/client_error.d.ts#L6)

Eslint didn't notify us of this duplicate condition.

**Comment by shazm** [comment]
Linter generally catch code-level bugs, prettier just unifies things like formatting. Prettier / ESLint tend to work pretty well together - tends to help with keeping a consistent "feel" to the code in terms of spacings, trailing comma's etc; while ESLint can ensure more code-specific things such as unused vars, imports etc.

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    semi: true,
+    trailingComma: 'all',
```
Comment (shazm) [inline .prettierrc.js:3]: The less characters the better I think?

**Review by shazm — APPROVED** [review]
I'm making the assumption the rest of the file changes are just prettier formatting fixes.

**Comment by enahum** [comment]
Well, our current linter rules have things about beautify the code like trailing commas, spacing, curly braces, etc.. that is why I'm not getting the need for this. But maybe you all can convince me otherwise. (And yes I know our new libs have prettier while every other js/ts project we have in the org does not)

**Comment by avinashlng1080** [comment]
conflicts with Mattermost ESLINT plugin.

**Pull request closed without merging**